lib/ex_doc/language/erlang.ex

```diff
@@ -265,7 +265,17 @@ defmodule ExDoc.Language.Erlang do
   end
 
   defp walk_doc({:code, attrs, [code], meta} = ast, config) when is_binary(code) do
-    case Autolink.url(code, :regular_link, config) do
+    ## Translate foo() to foo/0
```

Comment:
The feature is already there for types in Erlang ExDoc, I only expanded it to functions and callbacks for consistency.

That being said, I'm not commited to this functionality, tought I do think it makes some docs looks nicer to have `()` instead of `/0`.

Reply:
@garazdawi types are the only place where we traverse the AST to generate links, so I would argue this is a bit different. I would indeed go with `foo/0` for consistency. :)

Reply:
I'm not sure I follow the connection to traversing the AST? These are links from extra pages to types that take the form `` `t:sofs:type()` ``.

Reply:
Oh, I see. I misunderstood then. Shouldn't we update those and be consistent so we have `t:sofs:type/0`?

Reply:
I don't mind doing that, you can always write `` [`sofs:type()`](`t:sofs:type/0`) `` if you want the brackets.

Just be aware that doing that would be a backwards-incompatible change, albeit an undocumented behaviour, but there are tests verifying that it works.

Reply:
@garazdawi I would say it is _your_ decision. I am assuming this will be used as the foundation for future docs, so now would be the best time to start enforcing more consistent rules. I am fine with supporting your choice. :)